tests/main_test.go

```diff
@@ -0,0 +1,258 @@
+	t.Helper()
+
+	identitiesPath := filepath.Join(pebbleDir, identitiesFileName)
+	if err := os.WriteFile(identitiesPath, []byte(identitiesYAML), 0o755); err != nil {
+		t.Fatalf("Cannot create layers file: %v", err)
+	}
+}
+
+func pebbleRun(t *testing.T, pebbleDir string, args ...string) <-chan servicelog.Entry {
+	t.Helper()
+
+	logsCh := make(chan servicelog.Entry)
+
+	cmd := exec.Command("../pebble", append([]string{"run"}, args...)...)
+	cmd.Env = append(os.Environ(), "PEBBLE="+pebbleDir)
+
+	t.Cleanup(func() {
+		err := cmd.Process.Signal(os.Interrupt)
+		if err != nil {
+			t.Errorf("Error sending SIGINT/Ctrl+C to pebble: %v", err)
+		}
+		cmd.Wait()
+	})
+
+	stdoutPipe, err := cmd.StdoutPipe()
+	if err != nil {
+		t.Fatalf("Cannot create stdout pipe: %v", err)
+	}
+	stderrPipe, err := cmd.StderrPipe()
+	if err != nil {
+		t.Fatalf("Cannot create stderr pipe: %v", err)
+	}
+
+	err = cmd.Start()
+	if err != nil {
+		t.Fatalf("Error starting 'pebble run': %v", err)
+	}
+
+	done := make(chan struct{})
+	go func() {
+		defer close(logsCh)
+
+		readLogs := func(parser *servicelog.Parser) {
+			for parser.Next() {
+				if err := parser.Err(); err != nil {
+					t.Errorf("Cannot parse Pebble logs: %v", err)
+				}
+				select {
+				case logsCh <- parser.Entry():
+				case <-done:
+					return
+				}
+			}
+		}
+
+		// Both stderr and stdout are needed, because pebble logs to stderr
+		// while with "--verbose", services otuput to stdout.
+		stderrParser := servicelog.NewParser(stderrPipe, 4*1024)
+		stdoutParser := servicelog.NewParser(stdoutPipe, 4*1024)
+
+		// Channel to signal completion and close logsCh
+		done := make(chan struct{})
+		defer close(done)
+
+		go readLogs(stderrParser)
+		go readLogs(stdoutParser)
+
+		// Wait for both parsers to finish
+		<-done
+		<-done
+	}()
+
+	return logsCh
+}
+
+func waitForLogs(logsCh <-chan servicelog.Entry, expectedLogs []string, timeout time.Duration) error {
+	receivedLogs := make(map[string]struct{})
+
+	timeoutCh := time.After(timeout)
+	for {
+		select {
+		case log, ok := <-logsCh:
+			if !ok {
+				return errors.New("channel closed before all expected logs were received")
+			}
+
+			for _, expectedLog := range expectedLogs {
+				if _, ok := receivedLogs[expectedLog]; !ok && containsSubstring(log.Message, expectedLog) {
+					receivedLogs[expectedLog] = struct{}{}
+					break
+				}
+			}
+
+			allLogsReceived := true
+			for _, log := range expectedLogs {
+				if _, ok := receivedLogs[log]; !ok {
+					allLogsReceived = false
+					break
+				}
+			}
+
+			if allLogsReceived {
+				return nil
+			}
+
+		case <-timeoutCh:
+			missingLogs := []string{}
+			for _, log := range expectedLogs {
+				if _, ok := receivedLogs[log]; !ok {
+					missingLogs = append(missingLogs, log)
+				}
+			}
+			return errors.New("timed out waiting for log: " + strings.Join(missingLogs, ", "))
+		}
+	}
+}
+
+func containsSubstring(s, substr string) bool {
```

Comment:
Why is this needed? `containsSubstring` is actually longer to type than `strings.Contains`!